gemini/grounding/intro-grounding-gemini.ipynb

```diff
@@ -0,0 +1,623 @@
+    "\n",
+    "Note: The data store must be in the same project that you are using for Gemini.\n",
+    "\n",
+    "Once you've created a data store, obtain the Data Store ID and input it below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Replace this with your data store ID from Vertex AI Search\n",
+    "DATA_STORE_ID = \"your-data-store-id_1234567890123\"  # @param {type:\"string\"}\n",
+    "DATA_STORE_REGION = \"global\"  # @param {type:\"string\"}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now you can ask a question about object tables in BigQuery and when to use them:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PROMPT = \"When should I use an object table in BigQuery? And how does it store data?\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Text generation without grounding\n",
+    "\n",
+    "Make a prediction request to the LLM with no grounding:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response = model.generate_content(PROMPT)\n",
+    "\n",
+    "display(Markdown(response.candidates[0].text))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Text generation grounded in Vertex AI Search results\n",
+    "\n",
+    "Now we can add the `tools` keyword arg with a grounding tool of `grounding.VertexAISearch()` to instruct the LLM to first perform a search within your custom data store, then construct an answer based on the relevant documents:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datastore = f\"projects/{PROJECT_ID}/locations/{DATA_STORE_REGION}/collections/default_collection/dataStores/{DATA_STORE_ID}\"\n",
+    "tool = Tool.from_retrieval(\n",
+    "    grounding.Retrieval(grounding.VertexAISearch(datastore=datastore))\n",
+    ")\n",
+    "\n",
+    "response = model.generate_content(PROMPT, tools=[tool])\n",
+    "\n",
+    "display(Markdown(response.candidates[0].text))\n",
+    "\n",
+    "response"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the response without grounding only has limited information from the LLM about object tables in BigQuery that might not be accurate. Whereas the response that was grounded in Vertex AI Search results contains the most up to date information from the Google Cloud documentation about BigQuery, along with citations of the information."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "TpV-iwP9qw9c"
+   },
+   "source": [
+    "## Cleaning up\n",
```

Comment:
We probably also want to show the use of grounding with Gemini's chat modality.

Reply:
Ah, I forgot that chat is a different method. I can do that.